rfc/rfc022-discovery-service.md

```diff
@@ -74,40 +79,43 @@ The Verifiable Presentation MUST NOT be valid longer than the Verifiable Credent
 
 Clients MUST read the list by sending an HTTP GET to the server's service endpoint.
 The server MUST return a 200 OK response with a JSON object containing:
-- `entries` REQUIRED. MUST contain a JSON array with zero or more presentations.
-- `tag` REQUIRED. MUST be a JSON string referring to the last entry. The tag is an opaque value; the client SHOULD NOT derive any meaning from it.
+- `entries` REQUIRED. MUST contain a JSON object with a mapping of timestamp (as string) to presentation.
+- `timestamp` REQUIRED. MUST be a JSON integer equal to the timestamp of the last presentation.
+
+The timestamp definition used here is a lamport clock, which is a monotonically increasing integer.
 
-The `tag` query parameter MAY be used by the client to request a delta next time it reads the list.
-If no `tag` query parameter is provided, the server MUST return the full list.
+The `timestamp` query parameter MAY be used by the client to request a delta next time it reads the list.
+The server SHOULD return the presentations with a timestamp greater than the provided value.
+The server MUST return the latest `timestamp` value based on its internal state.
+If the query parameter is not provided, the server MUST return the full list.
+If based on the timestamp the server has no new presentations it returns an empty list for the `entries` object.
 Servers MUST only return the latest (valid) presentation per credential subject.
 
 Example:
 
 ``http request
-GET /list?tag=510 HTTP/1.1
+GET /list?timestamp=6 HTTP/1.1
 Content-Type: application/json
 
 {
-  "entries": [
-    "ey1234.etc.etc",
-    "ey5678.etc.etc"
-  ],
-  "tag": "515",
+  "entries": {
+    "7": "ey1234..."
+  }
+  "timestamp": 7
 }
 ``
 
 Clients MUST validate each presentation in the list as specified in section 4.
 If a presentation is valid, the client uses it in its system. If a presentation is not valid, it MUST be rejected.
 If one or more presentations are not valid, it SHOULD NOT reject the other presentations.
+Clients SHOULD update the `timestamp` value based on the value of the return object.
```

Comment:
"update the `timestamp` value" value of what?

Reply:
changed to `Clients SHOULD use the `timestamp` value in the return object for their next call`